test/inject.spec.tsx

```diff
@@ -365,4 +365,66 @@ describe('inject', () => {
 
   });
 
+  describe('via factory', () => {
+
+    class MyCustomWidgetOrg extends Composite {
+
+      service: MyServiceClass;
+      foo: string;
+      nonInjected: number;
+
+      constructor(
+        properties: Properties<Composite>,
+        @inject service: MyServiceClass,
+        @inject('foo') foo: string,
+        nothingToInject: number,
+        @inject('bar') implicitField: string
+      ) {
+        super(properties);
+        this.foo = foo;
+        this.service = service;
+        this.nonInjected = nothingToInject;
+      }
+
+    }
+
+    let MyCustomWidget: CallableConstructor<typeof MyCustomWidgetOrg>;
+    let widget: MyCustomWidgetOrg;
+
+    beforeEach(() => {
+      JSX.install(injector.jsxProcessor);
+      MyCustomWidget = asFactory(MyCustomWidgetOrg);
+      stringHandler = injection => String(injection.param);
+      widget = MyCustomWidget({
+        left: 3, top: 4, children: [Composite()]
+      });
+    });
+
+    it('injects parameterless', () => {
+      expect(widget.service).to.be.instanceOf(MyServiceClass);
+    });
+
+    it('injects with injection parameter', () => {
+      expect(widget.foo).to.equal('foo');
+    });
+
+    it('does not inject when not decorated', () => {
+      expect(widget.nonInjected).to.be.undefined;
+    });
+
+    it('injects implicit field', () => {
```

Comment:
looks duplicate test, can we merge it somehow?
and what do you mean by "implicit field"? you set `this.foo = foo` explicitly